File: data_structures/trees/post_order_traversal/python/postOrderTraversal.py
class Node:
    def __init__(self, val, left = None, right = None):
        self.val = val
        self.left = left
        self.right = right


def walk(curr, path):
    if curr == None:
        return path

    walk(curr.left, path)
    walk(curr.right, path)
    path.append(curr.val)

    return path

def postOrderTraversal(head):
    return walk(head, [])

if __name__ == "__main__": 
    left_node = Node(1)
    right_node = Node(5)
    bt_head = Node(4, left_node, right_node)
    left_node.left = Node(0)
    left_node.right = Node(2)
    right_node.left = Node(6)
    right_node.right = Node(7)

    print(postOrderTraversal(bt_head))

File: data_structures/trees/post_order_traversal/java/postOrderTraversal.java
import java.util.ArrayList; 

public class postOrderTraversal {

    public static ArrayList<Integer> walk(Node curr, ArrayList<Integer> path)
    {
        if(curr == null)
        {
            return path;
        }
        
        walk(curr.left, path);
        walk(curr.right, path);
        path.add(curr.val);

        return path;
    }

    public static void main(String[] args) {
        Node head = new Node(4);
        Node left = new Node(2);
        Node right = new Node(5);
        head.left = left;
        head.right = right;

        left.left = new Node(2);
        left.left.left = new Node(0);
        left.right = new Node(3);

        right.left = new Node(10);
        right.right = new Node(20);

        ArrayList<Integer> path = new ArrayList<Integer>();

        System.out.println(walk(head, path));
    }    
}
